orchagent/portsorch.cpp

```diff
@@ -470,6 +472,107 @@ void PortsOrch::updateDbPortOperStatus(sai_object_id_t id, sai_port_oper_status_
     }
 }
 
+bool PortsOrch::addPort(const set<int> &lane_set, uint32_t speed)
+{
+    SWSS_LOG_ENTER();
+
+    vector<uint32_t> lanes(lane_set.begin(), lane_set.end());
+
+    sai_attribute_t attr;
+    vector<sai_attribute_t> attrs;
+
+    attr.id = SAI_PORT_ATTR_SPEED;
+    attr.value.u32 = speed;
+    attrs.push_back(attr);
+
+    attr.id = SAI_PORT_ATTR_HW_LANE_LIST;
+    attr.value.u32list.list = lanes.data();
+    attr.value.u32list.count = static_cast<uint32_t>(lanes.size());
+    attrs.push_back(attr);
+
+    sai_object_id_t port_id;
+    sai_status_t status = sai_port_api->create_port(&port_id, gSwitchId, static_cast<uint32_t>(attrs.size()), attrs.data());
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to create port with the speed %u, rv:%d", speed, status);
+        return false;
+    }
+
+    m_portListLaneMap[lane_set] = port_id;
+
+    SWSS_LOG_NOTICE("Create port %lx with the speed %u", port_id, speed);
+
+    return true;
+}
+
+bool PortsOrch::removePort(sai_object_id_t port_id)
+{
+    SWSS_LOG_ENTER();
+
+    sai_status_t status = sai_port_api->remove_port(port_id);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to remove port %lx, rv:%d", port_id, status);
+        return false;
+    }
+
+    SWSS_LOG_NOTICE("Remove port %lx", port_id);
+
+    return true;
+}
+
+bool PortsOrch::initPort(const string &alias, const set<int> &lane_set)
+{
+    SWSS_LOG_ENTER();
+
+    /* Determine if the lane combination exists in switch */
+    if (m_portListLaneMap.find(lane_set) != m_portListLaneMap.end())
+    {
+        sai_object_id_t id = m_portListLaneMap[lane_set];
+
+        /* Determine if the port has already been initialized before */
+        if (m_portList.find(alias) != m_portList.end() && m_portList[alias].m_port_id == id)
+        {
+            SWSS_LOG_INFO("Port has already been initialized before alias:%s", alias.c_str());
+        }
+        else
+        {
+            Port p(alias, Port::PHY);
+
+            p.m_index = static_cast<int32_t>(m_portList.size()); // TODO: Assume no deletion of physical port
+            p.m_port_id = id;
+
+            /* Initialize the port and create router interface and host interface */
```

Comment:
>create router interface [](start = 39, length = 23)

this comment here is actualy incorrect, the initializePort no longer crate router interface.

Reply:
correct. let me modify it.